Cabal/src/Distribution/PackageDescription/Check/Monad.hs

```diff
@@ -0,0 +1,352 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Distribution.PackageDescription.Check.Monad
+-- Copyright   :  Francesco Ariis 2022
+-- License     :  BSD3
+--
+-- Maintainer  :  cabal-devel@haskell.org
+-- Portability :  portable
+--
+-- Primitives for package checking: check types and monadic interface.
+-- Having these primitives in a different module allows us to appropriately
+-- limit/manage the interface to suit checking needs.
+
+module Distribution.PackageDescription.Check.Monad
+                ( -- * Types and constructors
+                  CheckM(..),
+                  execCheckM,
+                  CheckInterface(..),
+                  CheckPackageContentOps(..),
+                  CheckPreDistributionOps(..),
+                  TargetAnnotation(..),
+                  PackageCheck(..),
+                  CheckExplanation(..),
+                  CEField(..),
+                  CEType(..),
+                  WarnLang(..),
+                  CheckCtx(..),
+                  pristineCheckCtx,
+                  initCheckCtx,
+                  PNames(..),
+
+                  -- * Operations
+                  ppPackageCheck,
+                  asksCM,
+                  localCM,
+                  checkP,
+                  checkPkg,
+                  liftInt,
+                  tellP,
+                  checkSpecVer
+
+                ) where
+
+import Distribution.Compat.Prelude
+import Prelude ()
+
+import Distribution.CabalSpecVersion (CabalSpecVersion)
+import Distribution.Package (packageName)
+import Distribution.PackageDescription.Check.Warning
+import Distribution.Simple.BuildToolDepends (desugarBuildToolSimple)
+import Distribution.Simple.Glob (Glob, GlobResult)
+import Distribution.Types.ExeDependency (ExeDependency)
+import Distribution.Types.GenericPackageDescription
+import Distribution.Types.LegacyExeDependency (LegacyExeDependency)
+import Distribution.Types.PackageId (PackageIdentifier)
+import Distribution.Types.PackageDescription (package, specVersion)
+import Distribution.Types.UnqualComponentName
+
+import qualified Control.Monad.Reader as Reader
+import qualified Control.Monad.Writer as Writer
+import qualified Control.Monad.Trans.Class as Trans
+import qualified Data.ByteString.Lazy as BS
+import qualified Data.Set as Set
+
+import Control.Monad
+
+-- ------------------------------------------------------------
+-- * Check monad
```

Comment:
Is this section header still relevant, given that there seems to be only one section?..

Reply:
And the whole module is called the same as the section.